osquery/events/windows/windows_event_log.cpp

```diff
@@ -108,43 +108,46 @@ unsigned long __stdcall WindowsEventLogEventPublisher::winEventCallback(
 
 Status WindowsEventLogEventPublisher::parseEvent(EVT_HANDLE evt,
                                                  pt::ptree& propTree) {
-  unsigned long buffSize = 0;
-  unsigned long buffUsed = 0;
-  unsigned long propCount = 0;
+  DWORD buffSize = 0;
+  DWORD buffUsed = 0;
+  DWORD propCount = 0;
   LPWSTR xml = nullptr;
-  auto ret = EvtRender(
-      nullptr, evt, EvtRenderEventXml, buffSize, xml, &buffUsed, &propCount);
-  if (ret == 0) {
-    if (GetLastError() == ERROR_INSUFFICIENT_BUFFER) {
+  Status status;
+
+  if (!EvtRender(nullptr,
+                 evt,
+                 EvtRenderEventXml,
+                 buffSize,
+                 xml,
+                 &buffUsed,
+                 &propCount)) {
+    if (ERROR_INSUFFICIENT_BUFFER == GetLastError()) {
       buffSize = buffUsed;
       xml = static_cast<LPWSTR>(malloc(buffSize));
-      EvtRender(nullptr,
-                evt,
-                EvtRenderEventXml,
-                buffSize,
-                xml,
-                &buffUsed,
-                &propCount);
-    } else {
-      return Status(GetLastError(), "Event rendering failed");
-    }
-  }
-  if (GetLastError() != ERROR_SUCCESS) {
-    if (xml != nullptr) {
-      free(xml);
-      xml = nullptr;
+      if (xml) {
+        EvtRender(nullptr,
+                  evt,
+                  EvtRenderEventXml,
+                  buffSize,
+                  xml,
+                  &buffUsed,
+                  &propCount);
+      } else {
+        status = Status(1, "Unable to reserve memory for event log buffer");
+      }
     }
-    return Status(GetLastError(), "Event rendering failed");
   }
-  std::stringstream ss;
-  ss << wstringToString(xml);
-  read_xml(ss, propTree);
 
-  if (xml != nullptr) {
-    free(xml);
-    xml = nullptr;
+  if (ERROR_SUCCESS == GetLastError()) {
+    std::stringstream ss;
+    ss << wstringToString(xml);
+    read_xml(ss, propTree);
+  } else {
+    status = Status(GetLastError(), "Event rendering failed");
   }
-  return Status(0, "OK");
+
+  free(xml);
```

Comment:
From my ttesting and research a free(nullptr) is a no-op, and I liked how clean this made the code. 

However, I'm starting to learn that in c++ you can never be to paranoid :).